lib/stripe/core_resources/payment_intent.ex

```diff
@@ -0,0 +1,294 @@
+defmodule Stripe.PaymentIntent do
+  @moduledoc """
+  Work with [Stripe `payment_intent` objects](https://stripe.com/docs/api/payment_intents).
+   You can:
+  - [Create a payment_intent](https://stripe.com/docs/api/payment_intents/create)
+  - [Retrieve a payment_intent](https://stripe.com/docs/api/payment_intents/retrieve)
+  - [Update a payment_intent](https://stripe.com/docs/api/payment_intents/update)
+  - [Confirm a payment_intent](https://stripe.com/docs/api/payment_intents/confirm)
+  - [Capture a payment_intent](https://stripe.com/docs/api/payment_intents/capture)
+  - [Cancel a payment_intent](https://stripe.com/docs/api/payment_intents/cancel)
+  - [List all payment_intent](https://stripe.com/docs/api/payment_intents/list)
+  """
+
+  use Stripe.Entity
+  import Stripe.Request
+  require Stripe.Util
+
+  @type last_payment_error :: %{
+          type: String.t(),
+          charge: String.t(),
+          code: String.t(),
+          decline_code: String.t(),
+          doc_url: String.t(),
+          message: String.t(),
+          param: String.t(),
+          source: Stripe.Card.t() | map
```

Comment:
Looks like `payment_intent` map (similar to checkout/session.ex) is in here as well.